app/src/main/java/com/github/ashutoshgngwr/noice/WakeUpTimerManager.kt

```diff
@@ -115,6 +117,39 @@ object WakeUpTimerManager {
     return withGson { it.fromJson(timer, Timer::class.java) }
   }
 
+  /**
+   * [savePresetID] saves the ID of the selected preset by the user
+   */
+  fun savePresetID(context: Context, selectedPresetID: String) {
+    val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
+    with(sharedPrefs.edit()) {
+      putString(SELECTED_PRESET_ID, selectedPresetID)
+      apply()
+    }
+  }
+
+  /**
+   * [removePresetIDFromSharedPrefs] remove the ID of the selected preset from Shared Preferences if exists.
+   */
+  fun removePresetIDFromSharedPrefs(context: Context, removePresetID: String) {
+    if (PreferenceManager.getDefaultSharedPreferences(context)
+        .getString("selectedPresetID", null) == removePresetID
+    ) {
+      with(PreferenceManager.getDefaultSharedPreferences(context).edit()) {
+        clear()
```

Comment:
I think you intended to use [`remove(...)`](https://developer.android.com/reference/android/content/SharedPreferences.Editor#remove(java.lang.String)) instead of [`clear()`](https://developer.android.com/reference/android/content/SharedPreferences.Editor#clear()). `clear()` will remove all values from the preferences.

Reply:
Oh yes, probably got confused when I looked up on documentation looking for a way to remove. Yes I meant to use `remove(...)`